src/libs/actions/IOU.js

```diff
@@ -445,7 +445,7 @@ function createSplitsAndOnyxData(participants, currentUserLogin, amount, comment
 
         // Regardless of the number of participants, we always want to push the iouReport update to onyxData
         optimisticData.push({
-            onyxMethod: existingOneOnOneChatReport ? CONST.ONYX.METHOD.MERGE : CONST.ONYX.METHOD.SET,
+            onyxMethod: ((hasMultipleParticipants && existingOneOnOneChatReport) || (!hasMultipleParticipants && existingGroupChatReport)) ? CONST.ONYX.METHOD.MERGE : CONST.ONYX.METHOD.SET,
```

Comment:
😬

Reply:
Yea, I'm not super happy about this. I'm open for other ideas on how to improve this code if you have any

Reply:
None for the moment, but maybe:
1. `((hasMultipleParticipants && existingOneOnOneChatReport) || (!hasMultipleParticipants && existingGroupChatReport))` but this in a variable so we can name it and that can help understanding what this case is, or
2. we can add a comment above

Reply:
added a comment

Reply:
From reading your comment and not the code yet.. could `((hasMultipleParticipants && existingOneOnOneChatReport) || (!hasMultipleParticipants && existingGroupChatReport))` be replaced by `existingGroupChatReport || existingOneOnOneChatReport`?

Reply:
`existingOneOnOneChatReport` will always be true if `!hasMultipleParticipants`, since it's the `groupChatReport`